.github/workflows/integrate.yaml

```diff
@@ -1,106 +1,79 @@
 name: CI
 
 on:
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
+  pull_request:
 
 jobs:
   lint:
-    name: Lint Check
+    name: Lint
     runs-on: ubuntu-latest
 
     steps:
     - name: Check out code
       uses: actions/checkout@v2
-
     - name: Install dependencies
-      run: sudo apt install python3-pip tox
-
+      run: |
+        sudo apt update
+        sudo apt install tox
     - name: Lint code
-      run: tox -e lint
+      run: tox -vve lint
 
   unit:
-    name: Unit Test
+    name: Unit Tests
     runs-on: ubuntu-latest
-
     steps:
-    - name: Check out code
-      uses: actions/checkout@v2
-
-    - name: Install dependencies
-      run: sudo apt install python3-pip tox
-
-    - name: Run unit tests
-      run: tox -e unit
-
-  deploy:
-    name: Deploy Test
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Install dependencies
+        run: |
+          pip install tox
```

Comment:
We can probably just inline one-liners like this

Reply:
Do you mean in-lining the checkout (eg: not having a `name:\n uses:` and instead just having `uses: actions/checkout@v2`), or changing the `run: |` to just `run: pip install tox`?

Reply:
The latter is what I meant specifically, not much point IMO in using the multiline string operator there.

In general, I also like just having `- run: pip install tox` instead of a name field that ends up obscuring the actual command run for steps that are just one single command anyways. That's out of scope for this PR, though